src/adapters/icon_manager.js

```diff
@@ -95,3 +95,14 @@ export function createMapGLIcon(imageFile, width, height) {
     img.src = imageFile;
   });
 }
+
+export function createPinIcon(iconOptions) {
+  const element = document.createElement('div');
+  element.innerHTML = `
+    <div
+      class="marker ${iconOptions.className || ''}"
+      ${iconOptions.disablePointerEvents && 'style="pointer-events:none;"'}
+    ></div>
+  `;
+  return element.firstElementChild;
```

Comment:
`return element` should be equivalent ?

Reply:
No, it would create useless `<div>`s in the map element, and the style doesn't apply the same… That's why you have the ugly hover effect in your gif below, because you fiddled with that :)
But I get what bothers you, it's that we declare a `<div>` only to fill up with innerHTML. I just used the same syntax as the similar existing function, but here I can rewrite it in a better way.